src/run-child-proc.cc

```diff
@@ -211,6 +211,10 @@ void base_process_service::run_child_proc(run_proc_params params) noexcept
             if (output_fd == -1) {
                 output_fd = open(logfile, O_WRONLY | O_CREAT | O_APPEND, S_IRUSR | S_IWUSR);
                 if (output_fd == -1) goto failure_out;
+                if (this->log_type == log_type_id::LOGFILE) {
```

Comment:
It was difficult for me to figure out why this check is necessary. I think it needs a comment ("if log type is NONE, we don't want to change ownership/permissions of /dev/null" or something similar).